cmd/rook/ceph/osd.go

```diff
@@ -274,6 +276,56 @@ func commonOSDInit(cmd *cobra.Command) {
 	clusterInfo.Monitors = mon.ParseMonEndpoints(cfg.monEndpoints)
 }
 
+func getLocation(clientset kubernetes.Interface, location string, topologyAware bool) (string, error) {
+	locArgs, err := client.FormatLocation(cfg.location, cfg.nodeName)
+	if err != nil {
+		return "", fmt.Errorf("invalid location. %+v", err)
+	}
+	// use zone/region/hostname labels in the crushmap
+	if cfg.topologyAware {
+		nodeName := os.Getenv(k8sutil.NodeNameEnvVar)
+		node, err := getNode(clientset, nodeName)
+		if err != nil {
+			return "", fmt.Errorf("topologyAware is set, but could not get the node: %+v", err)
+		}
+		nodeLabels := node.GetLabels()
+		// get zone
+		zone, ok := nodeLabels[corev1.LabelZoneFailureDomain]
+		if ok {
+			client.UpdateCrushMapValue(&locArgs, "zone", client.NormalizeCrushName(zone))
+		}
+		// get region
+		region, ok := nodeLabels[corev1.LabelZoneRegion]
+		if ok {
+			client.UpdateCrushMapValue(&locArgs, "region", client.NormalizeCrushName(region))
+		}
+
+	}
+
+	return strings.Join(locArgs, " "), nil
+}
+
+// getNode will try to get the node object for the provided nodeName
+// it will try using the node's name it's hostname label
+func getNode(clientset kubernetes.Interface, nodeName string) (*corev1.Node, error) {
+	var node *corev1.Node
+	var err error
+	// try to find by the node by matching the provided nodeName
+	node, err = clientset.CoreV1().Nodes().Get(nodeName, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		listOpts := metav1.ListOptions{LabelSelector: ""}
```

Comment:
Did you intend to use the hostname label to select? Looks like we will just query all nodes and select the first one if we didn't find the node in the first query.

Reply:
Yup. Good catch!